lib/webidl2.js

```diff
@@ -307,7 +335,10 @@
     }
 
     function type_suffix(obj) {
-      obj.nullable = !!consume("?");
+      const nullable = consume("?");
+      if (nullable) {
+        obj.nullable = { trivia: nullable.trivia };
```

Comment:
>Can you help me understand why we don’t default to false?

@marcoscaceres I had to determine where should information about optional "modifier" token go, and the result is this.

I could do `idlType.trivia.nullable === null`, but then we would have two fields about nullable token being absent. 

The `null` way keeps compatibility with `if (idlType.nullable)` and can also benefit from [future optional chaining syntax](https://github.com/tc39/proposal-optional-chaining/): `idlType.nullable?.trivia`.

Reply:
Thanks for the clarification, makes sense.